### ISSUE TITLE
net=host mode bug under mac, container with default bridge mode can not access host opening port 

### ISSUE THREAD

**Issue by ziyouchutuwenwu** [docker/for-mac#68]
**Output of `docker version`:**

```
(python2) mmcdeMacBook-Air:~ mmc$ docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:15:28 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 3
 Running: 2
 Paused: 0
 Stopped: 1
Images: 3
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 23
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host null bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.954 GiB
Name: moby
ID: D3CE:BLOM:7POQ:5KMA:LIMF:PSVT:MEAQ:GNEV:4JNX:N5YW:6HCJ:QD3X
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 26
 Goroutines: 40
 System Time: 2016-07-29T16:15:59.152792247Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
virtualbox 5.1.2 r108956
```

here is my host mac  network info
![local_network](https://cloud.githubusercontent.com/assets/7458713/17255309/402d0136-55eb-11e6-9651-988242db646a.png)

issue 1
**Steps to reproduce the issue:**
1. start a new container with -net=host mode

```
docker run -it --net=host edib/elixir-phoenix-dev /bin/bash
input ip address in container
```

**Describe the results you received:**

```
wish to see the same ip as host network interface eth0
```

**Describe the results you expected:**

```
host ip is 192.168.1.55, container ip is 192.168.65.xx, here is the screenshot
```

![net_host](https://cloud.githubusercontent.com/assets/7458713/17255588/739cfac0-55ec-11e6-96d3-7d4d869eaa72.png)

issue 2
**Steps to reproduce the issue:**
1. start a new container with default mode

```
docker run -it edib/elixir-phoenix-dev /bin/bash

in container run 
telnet to 172.17.0.1 8888
```

**Describe the results you received:**

```
8888 is my host port, listening for http request, screenshot
```

![web_page](https://cloud.githubusercontent.com/assets/7458713/17255487/15545b20-55ec-11e6-992f-937d1d981aec.png)

```
container telnet 172.17.0.1 failed, but  ping 172.17.0.1 runs ok
```

![default](https://cloud.githubusercontent.com/assets/7458713/17255537/430a1bcc-55ec-11e6-913e-d76c1ff7ef6a.png)

**Describe the results you expected:**

```
running container can telnet to host port, it runs ok under linux docker.
```

**Comment by darwink** [comment]
Any progress for this issue? We are meeting with the same issue.

**Comment by ziyouchutuwenwu** [comment]
hi, is there any time schedule? i checked it everyday, but nothing happened.

**Comment by seanhess** [comment]
I'm blocked by this issue as well.

**Comment by AlexGustafsson** [comment]
I'm experiencing the first issue as well.

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:04:48 2016
 OS/Arch:      darwin/amd64
 Experimental: true
```

**Comment by dave-tucker** [comment]
In this case, `--net=host` is working as expected... 
It's function however is not exactly intuitive on the mac ;)

The ability to set the `--net`, `--pid`, `--ipc` namespaces to `host` refers to `host` in the context of where the container engine is. In this case is a Linux VM. So for every use of `host`,  consider it to mean `vm`.

With that said, it would help us to understand what sort of applications need to run with `--net=host` so we can find a solution for this requirement.
You can always use port mapping as an alternative... 
For example,  `docker run -p 8888:8888` will allow you to access your service at `localhost:8888` from the Mac.

It might be easier to split this in to two issues to track being unable to access services running on the Mac from the container's default gateway IP address....

**Comment by AlexGustafsson** [comment]
@dave-tucker that clarifies things. Then how does one allow the container to access the mac host's interfaces (what host should mean imho)? Or is that a Linux only thing?

**Comment by djs55** [comment]
@AlexGustafsson: thanks for the clarification. Allowing the container to be accessed via a specific Mac host interface sounds entirely reasonable to me. I had hoped something like `docker run -p 10.0.0.1:80:80` would work, but unfortunately it fails because -- for historical reasons -- we try to bind to `10.0.0.1:80` on both the Mac and the VM, and unfortunately the VM doesn't have the same interface as the Mac. I'll investigate fixing this.

**Comment by djs55** [comment]
@ziyouchutuwenwu to telnet to a host port, you can telnet to the default gateway (`192.168.65.1`). For example I can run 
```
$ docker run -it justincormack/debian bash
Unable to find image 'justincormack/debian:latest' locally
latest: Pulling from justincormack/debian
efd26ecc9548: Pull complete
a3ed95caeb02: Pull complete
2df06b6623ba: Pull complete
Digest: sha256:4c0acbaf234244e7a565b1ed0f3bbe87561d1c0440f5e9382941eb35bb8e518a
Status: Downloaded newer image for justincormack/debian:latest
root@9833dddf1ed1:/# ssh user@192.168.65.1
The authenticity of host '192.168.65.1 (192.168.65.1)' can't be established.
ECDSA key fingerprint is cc:ce:e0:0f:03:ae:1b:be:b3:28:8b:75:40:8c:ff:e3.
Are you sure you want to continue connecting (yes/no)?
```
Is this sufficient for your use-case?

**Comment by ziyouchutuwenwu** [comment]
@djs55 ，thank you for your help
the actually ip gateway should be 192.168.1.1, 192.168.65.1 seems to be confused.
For users, displaying the same ip address will much better than the confused digital 65 i think.

**Comment by ziyouchutuwenwu** [comment]
not resolved for version
```
Version 1.12.1 (build: 12133)
2d5b4d9c3daa089e3869e6355a47dd96dbf39856
```

```
docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 10
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay bridge null host
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.20-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.953 GiB
Name: moby
ID: 57YZ:VNCF:YHUO:DG2H:67D2:TCYI:5FY7:FIZN:F4O7:XFAD:EGNN:3E4I
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 22
 Goroutines: 41
 System Time: 2016-09-16T21:44:52.575686599Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

```
docker version
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.7.1
 Git commit:   6f9534c
 Built:        Thu Sep  8 10:31:18 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 17:52:38 2016
 OS/Arch:      linux/amd64
```

my docker test command
```
docker run -it --net=host  edib/elixir-phoenix-dev /bin/bash
ip address | grep 192
inet 192.168.65.2/28 brd 192.168.65.15 scope global eth0
```

my host ip address is 
```
192.168.1.55
```

**Comment by justincormack** [comment]
@ziyouchutuwenwu the resolution discussed above was not being able to use `--net=host` which as discussed is difficult to achieve but being able to bind specifically to a particular host interface with port publishing. We may be able to make `--net=host` work one day but there is currently no timetable for this, it needs significant engineering work.

**Comment by ziyouchutuwenwu** [comment]
@justincormack thank you very much, i will use port mapping instead.

**Comment by roman-gliffy** [comment]
@ziyouchutuwenwu, I'm curious why its hard to bind to a host interface?

**Comment by michaeljs1990** [comment]
@justincormack thank you! I don't know why but this is the only answer I found in all the bugs on net host not currently working on docker for mac.

**Comment by torarnv** [comment]
I was very surprised to see this not working on the new shiny macOS version of Docker (working in the intuitive way, not in the "yes, strictly speaking 'host' means 'vm' on macOS").

Is there a workaround when `--net=host` is required for the container to do e.g. uPnP discovery of the local network? Eg: https://home-assistant.io/components/discovery/

**Comment by whoooami** [comment]
still doesn't work in 1.12.6.

**Comment by whoooami** [comment]
still doesn't work for version 1.12.6.

**Comment by lzomlot** [comment]
Any update on this issue. I tried the latest docker and I still experiencing it.

**Comment by manfredriem** [comment]
It appears it is possible for a standard xhyve install is able to allow access to the xhyve VM from the outside as indicated by the blog post at http://mifo.sk/post/xhyve-for-development

I think this would be the first step to properly supporting --net=host in Docker for Mac

Can someone from the Docker team investigate this?

**Comment by JockDaRock** [comment]
@justincormack you mentioned a while back this not having a timetable and needing significant engineering work... has there been any forward movement to put this on the timetable?

**Comment by abgalphabet** [comment]
i have similar issue but my problem is I got the 192.168.65.x ip as the manager node ip when running  "docker swarm init" which will not work for any worker node wanna join the swarm created.

I would need to provide the --advertise-addr <my real host ip> to make it working.

**Comment by kehao95** [comment]
If I didn't get it wrong, I was blocked by this issue. It's so confusing for a docker new user like me when I tried to run some HelloWorld level apps and Docker for Mac didn't respond as expected.
So please at lease make a note under the document about the 'host' mode before this issue is solved.

**Comment by jeantsai** [comment]
agree with kehao95, a note will help a lot and avoid frustrations from the production. Especially in the documentation of the awesome Mac native docker client.

**Comment by murugaratham** [comment]
I'm facing this issue as well, I'm using emby server but the dlna is publishing on the internal subnet only, +1 for timetable

**Comment by newdark** [comment]
Here is something as a work around to this problem. Right now there is an issue open(#155) that addresses this problem. From reading through most of the comments online I found that this is currently a missing feature that is available in Linux. The best solution I found to get around the problem was to run in bridged mode and change everything to [docker.for.mac.localhost](https://docs.docker.com/docker-for-mac/networking/#use-cases-and-workarounds)

Here is example docker that I was using to test out with curl commands
```shell
$  docker run -d --name elasticsearch -p 9200:9200 elasticsearch:1.4.5
$ curl http://localhost:9200
{
  "status" : 200,
  "name" : "Slither",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "1.4.5",
    "build_hash" : "2aaf797f2a571dcb779a3b61180afe8390ab61f9",
    "build_timestamp" : "2015-04-27T08:06:06Z",
    "build_snapshot" : false,
    "lucene_version" : "4.10.4"
  },
  "tagline" : "You Know, for Search"
}
$ curl http://docker.for.mac.localhost:9200
curl: (6) Could not resolve host: docker.for.mac.localhost
```
The reason for this error above is because I am not inside my docker image and as a result that domain name is not defined. The good news is that you can get aroudn that problem by just adding it to your `/etc/hosts`.

```
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1	localhost
127.0.0.1 docker.for.mac.localhost
255.255.255.255	broadcasthost
::1             localhost
```
Now the curl command should work with issue
```shell
$ curl http://docker.for.mac.localhost:9200
{
  "status" : 200,
  "name" : "Slither",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "1.4.5",
    "build_hash" : "2aaf797f2a571dcb779a3b61180afe8390ab61f9",
    "build_timestamp" : "2015-04-27T08:06:06Z",
    "build_snapshot" : false,
    "lucene_version" : "4.10.4"
  },
  "tagline" : "You Know, for Search"
}
```
Here is how to setup the docker image to test it out for your self. 

```shell
$ docker run --rm -it nginx /bin/bash
$ apt-get update
$ apt-get install curl
$ curl http://docker.for.mac.localhost:9200
```
Until the `--network=host` starts working this seems to be my solution. Hope it can help here. Oh, also something to note I am not using Virtual Box on my mac. https://store.docker.com/editions/community/docker-ce-desktop-mac

Also as a piece of information, I am running docker Version 17.06.2-ce-mac27 (19124)

**Comment by VinceXie** [comment]
2018 still not work in mac.

**Comment by rainbow-alex** [comment]
We need `--network host` to run a selenium container which can access a non-docker server on the host.

**Comment by hanrok** [comment]
Still not working, does anybody solve it in some workaround?

**Comment by rahonalab** [comment]
same issue here, still doesn't work

**Comment by sfragata** [comment]
I think this problem must have at least 2 years and there is no move from Docker....
sad

**Comment by buzzdan** [comment]
also need it working...

**Comment by RSamal** [comment]
I think the reason is xhyve, as docker on mac run all the container under this little VM. I just ran using the --net=host and the network interfaces in xhyve VM and in container matches but not with the original host which is macos. so it makes sense why --net=host does not work on mac as the actual host is xhyve in this case, not the mac.  

To login into xhyve 
`screen ~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty`

And here are the network interfaces in the xhyve vm
`linuxkit-025000000001:~# ifconfig
br-47d654cdc4b8 Link encap:Ethernet  HWaddr 02:42:E3:30:A4:78
          inet addr:10.0.0.1  Bcast:10.0.0.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
	options=60<TSO4,TSO6>
	ether 6a:00:01:71:ab:90
docker0   Link encap:Ethernet  HWaddr 02:42:FE:10:5C:44
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:feff:fe10:5c44/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:718 (718.0 B)

docker_gwbridge Link encap:Ethernet  HWaddr 02:42:D4:6B:46:79
          inet addr:172.20.0.1  Bcast:172.20.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:d4ff:fe6b:4679/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:20 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:1488 (1.4 KiB)

eth0      Link encap:Ethernet  HWaddr 02:50:00:00:00:01
          inet addr:192.168.65.3  Bcast:192.168.65.255  Mask:255.255.255.0
          inet6 addr: fe80::3d54:c71b:fbef:5f7b/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:26825 errors:0 dropped:0 overruns:0 frame:0
          TX packets:20154 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:19936409 (19.0 MiB)  TX bytes:2268037 (2.1 MiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:10362 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10362 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:945027 (922.8 KiB)  TX bytes:945027 (922.8 KiB)

veth2f727a0 Link encap:Ethernet  HWaddr 86:35:3E:23:15:16
          inet6 addr: fe80::8435:3eff:fe23:1516/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:39 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:2906 (2.8 KiB)`

and a container with --net=host 
`➜  ~ docker run -it --net=host alpine ash
/ # ifconfig
br-47d654cdc4b8 Link encap:Ethernet  HWaddr 02:42:E3:30:A4:78
          inet addr:10.0.0.1  Bcast:10.0.0.255  Mask:255.255.255.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

docker0   Link encap:Ethernet  HWaddr 02:42:FE:10:5C:44
          inet addr:172.17.0.1  Bcast:172.17.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:feff:fe10:5c44/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:718 (718.0 B)

docker_gwbridge Link encap:Ethernet  HWaddr 02:42:D4:6B:46:79
          inet addr:172.20.0.1  Bcast:172.20.255.255  Mask:255.255.0.0
          inet6 addr: fe80::42:d4ff:fe6b:4679/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:20 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:1488 (1.4 KiB)

eth0      Link encap:Ethernet  HWaddr 02:50:00:00:00:01
          inet addr:192.168.65.3  Bcast:192.168.65.255  Mask:255.255.255.0
          inet6 addr: fe80::3d54:c71b:fbef:5f7b/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:26825 errors:0 dropped:0 overruns:0 frame:0
          TX packets:20154 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:19936409 (19.0 MiB)  TX bytes:2268037 (2.1 MiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:10658 errors:0 dropped:0 overruns:0 frame:0
          TX packets:10658 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:971963 (949.1 KiB)  TX bytes:971963 (949.1 KiB)

veth2f727a0 Link encap:Ethernet  HWaddr 86:35:3E:23:15:16
          inet6 addr: fe80::8435:3eff:fe23:1516/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:39 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:2906 (2.8 KiB)`

So the network interfaces match exactly and I can access the container IP and port from the xhyve. Not from mac.

**Comment by torarnv** [comment]
@RSamal the fact that docker on macOS is implemented via an xhyve VM is an "implementation detail", and when I as a user pass `--net host`, I expect the same behavior as if as I was running the docker client tool on a Linux host.
> unless we do some port forwarding from the hyperviser to the original host.

Exactly. This is the kind of magic that docker should do in the background, to ensure that things work the same regardless of which platform docker runs on.

**Comment by diegolaciar** [comment]
+1
same issue here.
does anybody solve it in some workaround?

**Comment by jiminssy** [comment]
My workaround has been using the local address "host.docker.internal" as it was mentioned here.
https://docs.docker.com/docker-for-mac/networking/#httphttps-proxy-support

**Comment by JeffCorract** [comment]
Any news whether this issue is still open?

**Comment by loblab** [comment]
Seems no solution yet?
In some case, we cannot use port map (like device discovery). host mode is useful, at least, for debugging.

**Comment by BenAndDevelopment** [comment]
Any updates on this?

**Comment by snky1987** [comment]
4 years and still no response/fix?

**Comment by vrawal-qeexo** [comment]
??

**Comment by ethaniel** [comment]
A temporary workaround is to install a ubuntu VM through VirtualBox, hook it up via bridge mode and THEN run the docker inside of it.

**Comment by dbux** [comment]
[This response](https://github.com/docker/for-mac/issues/2716#issuecomment-456233826) to a similar issue worked great for me. Moving from VirtualBox to xhyve was clearly a regression for networking in Docker.

**Comment by cumanzor** [comment]
Absolutely horrifying that in 2021 this still doesn't work, and yet they still push you to "upgrade" to the next tier whenever there's a new update.

**Comment by ganttee** [comment]
[Running HomeBridge on Docker without Host Network Mode](https://www.devwithimagination.com/2020/02/02/running-homebridge-on-docker-without-host-network-mode/)

**Comment by edgariscoding** [comment]
> Running HomeBridge on Docker without Host Network Mode

@ganttee have you tested this?

**Comment by Tonyelo** [comment]
Hi everyone!, I tried to follow:

 https://www.devwithimagination.com/2020/02/02/running-homebridge-on-docker-without-host-network-mode/

However, I got lost on the generate_sevice.sh script. Nor I understand the avahi side of things. I think I missing a few steps.

I am trying to run it on a Mac M1 silicon. Any help would appreciated. thanks!

**Comment by HunterDG** [comment]
Anyone know if macvlan interfaces are supported on Macs?  (I assume not)

**Comment by ShutdownRepo** [comment]
It's been almost 6 years now that this issue is up. Isn't a doable thing for Docker team to implement a dynamic port forwarding feature between xhyve and the host in order to have `--network host` work as intended (in the user's perspective, no the dev's)?

**Comment by ThePragmaticArt** [comment]
Sort of baffled I just wasted hours debugging an issue that has existed for years...

**Comment by kopax-polyconseil** [comment]
Same here, but I still hope I will find a workaround, I've read some people succeed to configure a VM and use host network mode properly, not sure how to do though a tutorial would be more than welcomed. 
Use case : use a macbook air as a node in a cloud cluster.

**Comment by HunterDG** [comment]
Was previously boot2docker/docker-machine.  Guessing Rancher Desktop or Colima now. 

Perhaps someone will correct me.  

https://www.paolomainardi.com/posts/docker-performance-macos/

**Comment by gabo-magnet** [comment]
> Now with the [new release 4.14.0](https://docs.docker.com/desktop/release-notes/#4140) of Docker Desktop on Mac utilizing the Apple Virtualization Framework, it should be possible to add another network adapter featuring the bridged network aka. network_mode = host ?
> 
> https://developer.apple.com/documentation/virtualization/vzvirtualmachineconfiguration/3656724-networkdevices
> 
> Is this a big challenge? Can somebody point out the hypervisor configuration for the Apple Virtualization Framework here? :)
> 
> Also thanks for putting it in the "Considering" space of the docker-roadmap ! 👍

https://github.com/docker/roadmap/issues/238#issuecomment-1328246842

**Comment by nicks** [comment]
going to mark this as a duplicate of https://github.com/docker/roadmap/issues/238 so we have one place to track this.
